### PR TITLE
Added custom path ability

### PR DESCRIPTION
**Pull request by ashleybrgr**
Allows the option to specify a custom bundle path as per git issue #131. Supports dependencies using Swift Package Manager since the package manager does not have the ability to copy over dependency resources.

### CODE REVIEW

**Comment by efirestone** [comment]
I'm not a maintainer of the project, but the diff looks good to me (and we're interested in getting this change as well).

A possible improvements for consideration:
- Define `JSONDataCallback` as `() throws -> Data` rather than `() -> Data?`, with the thought being that a failure to get the data is really an error (there is no "successful" case where no data is returned). This is mostly just an API clarification.
- Rather than defining `MetadataManager.init` as:
```
public init (JSONDataCallback: JSONDataCallback? = nil)
```
instead define it as
```
public init (JSONDataCallback: JSONDataCallback = defaultDataCallback)
```
where `defaultDataCallback` does the standard behavior of looking up the data from the local bundle. This has two small advantages:
- The API is clearer. There is no guessing what `nil` means in the API.
- There is no internal dependency on the local bundle's file. `defaultDataCallback` would be a static function, so after initialization the class has zero notion of the default bundle and its resources.

**Comment by efirestone** [comment]
@marmelroy, any chance of landing this soon? We'd love to be able to adopt this project but require this addition.

File: PhoneNumberKit/MetadataManager.swift [hunk 1]
```diff
@@ -9,19 +9,19 @@
...
-    
+
     // MARK: Lifecycle
 
     /**
      Private init populates metadata territories and the two hashed dictionaries for faster lookup.
      */
-    public init () {
-        territories = populateTerritories()
+     public init (metadataPath: String? = nil) {
```
Comment (phoney) [inline PhoneNumberKit/MetadataManager.swift]: I recommend that the metadataPath is a file URL not a String. A String can be anything. You can validate that the URL is a fileURL. The code doesn't change much but using a URL here would make the intention more clear and probably reduce some user bugs.

**Comment by marmelroy** [comment]
Thanks all and sorry for the delay with this. I think this is a great improvement and will add it to the next release.

**Pull request merged**